modules/REST/V2/Languages.php

```diff
@@ -0,0 +1,737 @@
+						'type'        => 'string',
+					),
+				),
+				array(
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'get_item' ),
+					'permission_callback' => '__return_true',
+					'args'                => array(
+						'context' => $this->get_context_param( array( 'default' => 'view' ) ),
+					),
+				),
+				array(
+					'methods'             => WP_REST_Server::EDITABLE,
+					'callback'            => array( $this, 'update_item' ),
+					'permission_callback' => array( $this, 'update_item_permissions_check' ),
+					'args'                => $this->get_endpoint_args_for_item_schema( WP_REST_Server::EDITABLE ),
+				),
+				array(
+					'methods'             => WP_REST_Server::DELETABLE,
+					'callback'            => array( $this, 'delete_item' ),
+					'permission_callback' => array( $this, 'delete_item_permissions_check' ),
+				),
+				'schema' => array( $this, 'get_public_item_schema' ),
+			)
+		);
+	}
+
+	/**
+	 * Retrieves a collection of languages.
+	 *
+	 * @since 3.7
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
+	 *
+	 * @phpstan-template T of Fields
+	 * @phpstan-param WP_REST_Request<T> $request
+	 */
+	public function get_items( $request ) {
+		$languages = array();
+
+		foreach ( $this->model->languages_model->get_languages_list() as $language ) {
+			$languages[] = $this->prepare_item_for_response( $language, $request );
+		}
+
+		return rest_ensure_response( $languages );
+	}
+
+	/**
+	 * Creates one language from the collection.
+	 *
+	 * @since 3.7
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
+	 *
+	 * @phpstan-template T of Fields
+	 * @phpstan-param WP_REST_Request<T> $request
+	 */
+	public function create_item( $request ) {
+		if ( ! empty( $request['id'] ) ) {
+			return new WP_Error(
+				'rest_exists',
+				__( 'Cannot create existing language.', 'polylang' ),
+				array( 'status' => 400 )
+			);
+		}
+
+		/**
+		 * $request: name, code, locale, direction, order,      flag
+		 * $args:    name, slug, locale, rtl,       term_group, flag? (from add_language())
+		 */
+		if ( ! isset( $request['name'], $request['code'], $request['direction'], $request['flag'] ) ) {
+			$languages = include POLYLANG_DIR . '/settings/languages.php';
+
+			if ( empty( $languages[ $request['locale'] ] ) ) {
+				return new WP_Error(
+					'rest_invalid_locale',
+					__( 'The locale is invalid.', 'polylang' ),
+					array( 'status' => 400 )
+				);
+			}
+
+			$language = $languages[ $request['locale'] ];
+			$defaults = array(
+				'name'       => $language['name'],
+				'slug'       => $language['code'],
+				'rtl'        => (int) ( 'rtl' === $language['dir'] ),
+				'flag'       => $language['flag'],
```

Comment:
I wonder if `flag` should be a proper path to a custom flag here.
Maybe that's something to discuss with the other devs.